python/cuml/cuml/accel/__main__.py

```diff
@@ -13,128 +13,206 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from __future__ import annotations
 
+import argparse
 import code
-import pickle
 import runpy
 import sys
+from textwrap import dedent
+
+from cuml.accel.core import install
+from cuml.accel.estimator_proxy_mixin import ProxyMixin
+from cuml.internals import logger
+
+
+def execute_source(source: str, filename: str = "<stdin>") -> None:
+    """Execute source the same way python's interpreter would.
+
+    source: str
+        The source code to execute.
+    filename: str, optional
+        The filename to execute it as. Defaults to `"<stdin>"`
+    """
+    try:
+        exec(compile(source, filename, "exec"))
+    except SystemExit:
+        raise
+    except BaseException:
+        typ, value, tb = sys.exc_info()
+        # Drop our frame from the traceback before formatting
+        sys.__excepthook__(
+            typ,
+            value.with_traceback(tb.tb_next),
+            tb.tb_next,
+        )
+        # Exit on error with the proper code
+        code = 130 if typ is KeyboardInterrupt else 1
+        sys.exit(code)
+
+
+def parse_args(argv: list[str]) -> argparse.Namespace:
+    """Parse args for the `cuml.accel` CLI."""
+    parser = argparse.ArgumentParser(
+        prog="python -m cuml.accel",
```

Comment:
Do we have to put an explicit value here? Asking because there are people who use `python3`, `python3.12`, `ipython` (and who knows what else) for who it would be nice to have "the right thing" displayed here. But maybe it isn't possible to do?

Reply:
We do, otherwise python defaults to the filename (`__main__.py` in this case). I _think_ all (or at least most) distributions use a `python` symlink along with `python3`, so `python -m ...` should be fine IMO. The only other thing I could think of is stripping the exec name off of `sys.executable`, but that would add an unnecessary version suffix on most linux distros (`python3.13 -m cuml.accel` instead of the valid `python -m cuml.accel`). Since this only shows up in help messages, I'd opt to do the simple thing and keep things as they are.